modes/vterm/evil-collection-vterm.el

```diff
@@ -282,8 +299,15 @@ But don't allow the cursor to move bellow the last prompt line."
     "G" 'vterm-reset-cursor-point)
 
   (evil-collection-define-key 'visual 'vterm-mode-map
+    "x" 'evil-collection-vterm-delete-backward-char)
+
+  (evil-collection-define-key '(normal visual) 'vterm-mode-map
     "d" 'evil-collection-vterm-delete
-    "x" 'evil-collection-vterm-delete-backward-char))
+    "l" 'evil-collection-vterm-forward-char
+    "h" 'evil-collection-vterm-backward-char
```

Comment:
One last thing, can you add a flag (can default to true), to conditionally enable this.

Something like:

if flag:
 bind l h $ ^ to -> new feature
else:
 bind ^ (whatever other key needs to be deleted from thsi CL)

This cl is moving some lines to other parts so hard to tell what's missing/gained easily. (Not taking too close a look)

Reply:
I agree, that sounds like a good idea. Please take a look.

Reply:
I think we should create 2 separate flags (one for ^ $ and one for h l) and then check the flags only in evil-collection-vterm-setup to conditionally bind the functions.

One use case I'm thinking (which is one I'd be using) is to only bind this for h & l movements but keep ^ and $ movements alone.

Reply:
> I think we should create 2 separate flags (one for ^ $ and one for h l)
> One use case I'm thinking (which is one I'd be using) is to only bind this for h & l movements but keep ^ and $ movements alone.

It doesn't really make sense to me, separating the configuration into two options. In my case, I either want the prompt to behave like one long line or not :-). That being said, if you are interested in having this behavior just for h, k and not ^, $, maybe others will as well. So sure, no problem with me. Do you have any preference for the option names?

> check the flags only in evil-collection-vterm-setup to conditionally bind the functions.

I am trying to find some other key bind that is done this way but not seeing any. Have you seen this approach somewhere else in `evil-collection`? I'd like to be consistent with it.

Reply:
*bump* @jojojames

Reply:
https://github.com/emacs-evil/evil-collection/blob/5092f5e04cfd0a31b1c5c46cb6b792a9e917c728/modes/elpaca/evil-collection-elpaca.el#L105

Reply:
It won't be as easy as we thought. Or at least as I thought.

We would have to define two variants for every movement. I am proposing `hj^$` in this PR but I forgot `X`, `0`, operations till the end of the line like `C`, `S`, `D`, `Y`, jumps like `t<char>`, `T<char>`, or line operations like `yy`, `dd`, and whatever else. That would be too many vterm specific key binds. But more importantly too many key binds for this line-continuation feature. Imagine how much would the number of key binds blow up if we want to add some other feature with conditional keybinds. There must be a smarter way of doing this but I am struggling to find it.

I am playing with overriding something like `evil-line-expand` but without much success. It may be a dead end as well.

You have much more experience with Evil than me, do you have any idea @jojojames, please?

Reply:
I think it'd be better to group a set of binds together than to consider them each individually, that way the amount of new flags won't balloon out of control. h&j would be in a separate group from $&0 and whatever else.

The grouping can be decided reasonably, I'd assume.